netty/src/test/java/io/grpc/transport/netty/NettyClientHandlerTest.java

```diff
@@ -249,68 +279,67 @@ public void inboundDataShouldForwardToStream() throws Exception {
     verify(stream).transportDataReceived(eq(content), eq(false));
   }
 
   @Test
-  public void createShouldQueueStream() throws Exception {
-    // Disallow stream creation to force the stream to get added to the pending queue.
+  public void createShouldBufferStream() throws Exception {
+    // Force the stream to be buffered.
     setMaxConcurrentStreams(0);
-    handler.write(ctx, new CreateStreamCommand(grpcHeaders, stream),
-        promise);
-
-    // Make sure the write never occurred.
-    verify(frameListener, never()).onHeadersRead(eq(ctx),
-        eq(3),
-        any(Http2Headers.class),
-        eq(0),
-        eq(Http2CodecUtil.DEFAULT_PRIORITY_WEIGHT),
-        eq(false),
-        eq(0),
-        eq(false));
+    handler.write(ctx, new CreateStreamCommand(grpcHeaders, stream), promise);
+    verifyNoMoreInteractions(ctx);
   }
 
   @Test
-  public void receivedGoAwayShouldFailQueuedStreams() throws Exception {
-    // Force a stream to get added to the pending queue.
+  public void receivedGoAwayShouldFailBufferedStream() throws Exception {
+    // Force the stream to be buffered.
     setMaxConcurrentStreams(0);
-    handler.write(ctx, new CreateStreamCommand(grpcHeaders, stream),
-        promise);
-
+    ChannelPromise promise = new DefaultChannelPromise(channel, ImmediateEventExecutor.INSTANCE);
+    handler.write(ctx, new CreateStreamCommand(grpcHeaders, stream), promise);
     handler.channelRead(ctx, goAwayFrame(0));
-    verify(promise).setFailure(any(Throwable.class));
+    assertTrue(promise.isDone());
+    assertFalse(promise.isSuccess());
+    verify(stream).transportReportStatus(any(Status.class), eq(true),
+            notNull(Metadata.Trailers.class));
   }
 
   @Test
   public void receivedGoAwayShouldFailUnknownStreams() throws Exception {
-    // Force a stream to get added to the pending queue.
-    handler.write(ctx, new CreateStreamCommand(grpcHeaders, stream),
-        promise);
+    handler.write(ctx, new CreateStreamCommand(grpcHeaders, stream), promise);
 
     // Read a GOAWAY that indicates our stream was never processed by the server.
-    handler.channelRead(ctx,
-        goAwayFrame(0, 8 /* Cancel */, Unpooled.copiedBuffer("this is a test", UTF_8)));
+    handler.channelRead(ctx, goAwayFrame(0, 8 /* Cancel */,
+            Unpooled.copiedBuffer("this is a test", UTF_8)));
     ArgumentCaptor<Status> captor = ArgumentCaptor.forClass(Status.class);
-    InOrder inOrder = inOrder(stream);
-    inOrder.verify(stream, calls(1)).transportReportStatus(captor.capture(), eq(false),
-        notNull(Metadata.Trailers.class));
+    verify(stream).transportReportStatus(captor.capture(), eq(false),
+            notNull(Metadata.Trailers.class));
     assertEquals(Status.CANCELLED.getCode(), captor.getValue().getCode());
     assertEquals("HTTP/2 error code: CANCEL\nthis is a test",
-        captor.getValue().getDescription());
+            captor.getValue().getDescription());
```

Comment:
It also seems like your indentation in the IDE is set too high. Here the 4 spaces were already enough, and you didn't change the line for any other reason. I don't actually know why it is now indented 8. I could easily want to send out a later PR that just reverts this change.